### DISCUSSION TITLE
Rich text support

### DISCUSSION THREAD

**Discussion by onethingwelldev** [snarfed/bridgy-fed#771] [General]
This is from BF docs:
> Mastodon preserves HTML links and line breaks, but removes all other formatting and tags. Other fediverse sites vary in their HTML handling.

Since v4.2, Mastodon supports some additional tags (https://docs.joinmastodon.org/spec/activitypub/#sanitization), so I tried to include `<pre>` tag inside `h-entry`, but it seems to be stripped out by BF. Do I need to do something else to include this?

**Comment by snarfed** [comment]
Thanks for the nudge! That section of the BF docs should indeed be updated.

I haven't tested interop with Mastodon 4.2's formatting support much yet, but BF is definitely preserving your HTML content. Here's the activity it sent for one of your test posts that included `<pre>`:

```json
{
  "@context": "https://www.w3.org/ns/activitystreams",
  "type": "Create",
  "id": "https://fed.brid.gy/r/https://onethingwell.dev/test2#bridgy-fed-create",
  "actor": "https://fed.brid.gy/onethingwell.dev",
  "published": "2023-12-23T16:24:53.247080+00:00",
  "object": {
    "type": "Note",
    "id": "https://fed.brid.gy/r/https://onethingwell.dev/test2",
    "content": "Test...\n    <pre>test code</pre>",
    "url": "https://fed.brid.gy/r/https://onethingwell.dev/test2",
    "attributedTo": "https://fed.brid.gy/onethingwell.dev",
    "contentMap": {"en": "Test...\n    <pre>test code</pre>"},
    "to": ["https://www.w3.org/ns/activitystreams#Public"]
  },
  "to": ["https://www.w3.org/ns/activitystreams#Public"]
}
```

**Reply by onethingwelldev** [reply]
Ah, okay, so it's definitely something on Mastodon's side... Maybe I need to include something else, because I've seen posts from other instances with these tags in my feed, but it was removed from this one for some reason.
